src/util/pubkey_finder/email_sigs_gcd.py

```diff
@@ -86,6 +191,68 @@ async def find_key_for_signature_pair(dsp: Dsp, sig1: EmailSignature, sig2: Emai
 		    'timestamp': datetime.now(),
 		})
 
+async def check_for_matching_key_period(dsp: Dsp, sig1: EmailSignature, sig2: EmailSignature):
+	"""
+	Check if two email signatures' timestamps fall within known key periods from archive.prove.email.
+	
+	Args:
+			dsp (Dsp): Domain/selector pair object containing domain and selector
+			sig1 (EmailSignature): First email signature to check
+			sig2 (EmailSignature): Second email signature to check
+			
+	Returns:
+			bool: True if GCD calculation should proceed (no matching key period found),
+						False if matching key period was found and GCD can be skipped
+	"""
+	# Check if timestamps fall within known key periods from archive.prove.email
+	async with httpx.AsyncClient() as client:
+		response = await client.get(f"https://archive.prove.email/api/key?domain={dsp.domain}")
+		timestamp_1_covered, timestamp_2_covered = False, False
+		if response.status_code == 200:
+			keys = response.json()
+			matching_keys = [k for k in keys if k["selector"] == dsp.selector]
+					
+			for key in matching_keys:
+				first_seen = datetime.fromisoformat(key["firstSeenAt"].replace("Z", "+00:00"))
+				last_seen = datetime.fromisoformat(key["lastSeenAt"].replace("Z", "+00:00"))
+				
+				if sig1.timestamp and first_seen <= sig1.timestamp <= last_seen:
+					timestamp_1_covered = True
+					try:
+						logging.info(f"Validating correct signature {sig1.id} with key {key['keyData']}")
```

Comment:
seems that `key` doesn't have a field called keyData, 
it only has what is included here: https://github.com/zkemail/archive.zk.email/blob/main/src/app/api/key/route.ts#L31

so we will always get an exception here